Deprecate Develocity plugin versions before 4.0 (#37742)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_9.adoc

=== Deprecations

[[deprecated_develocity_plugin_pre_4_0]]

> ==== Deprecation of Develocity plugin versions before 4.0

Reviewer: Is this essentially the same message (and the same code changes) as we did when deprecating the Develocity plugin in Gradle 8.x? It doesn't seem to be the case. This is what we currently have in upgrading_version_8.adoc:

```
[[unsupported_ge_plugin_3.13]]
==== Earliest supported Develocity plugin version is 3.13.1

Starting in Gradle 9.0.0, the earliest supported Develocity plugin version is 3.13.1.
The plugin versions from 3.0 up to 3.13 will be ignored when applied.

Upgrade to version 3.13.1 or later of the Develocity plugin.
You can find the link:https://plugins.gradle.org/plugin/com.gradle.enterprise[latest available version on the Gradle Plugin Portal].
More information on the compatibility can be found link:https://docs.gradle.com/enterprise/compatibility/#build_scans[here].
```

I would expect a very similar message, with updated links.

=== Deprecations

[[deprecated_develocity_plugin_pre_4_0]]

> ==== Earliest supported Develocity plugin version starting in Gradle 10 is 4.0

Reviewer 2: ```suggestion
==== Develocity plugin versions before 4.0 are no longer supported in Gradle 10
```

> Starting in Gradle 10, the earliest supported Develocity plugin version is 4.0.

Reviewer 2: ```suggestion
Gradle 10 requires Develocity plugin 4.0 or later.
```

> The plugin versions before 4.0 rely on Gradle's implicit parent-project property lookup, which is being removed in Gradle 10, and will be ignored when applied.

Reviewer 2: ```suggestion
Earlier versions rely on Gradle's implicit parent-project property lookup, which will be removed in Gradle 10 and silently ignored if applied.
```

> Upgrade to version 4.0 or later of the Develocity plugin.

Reviewer 2: ```suggestion
```

> You can find the link:https://plugins.gradle.org/plugin/com.gradle.develocity[latest available version on the Gradle Plugin Portal].

Reviewer 2: ```suggestion
Upgrade to the link:https://plugins.gradle.org/plugin/com.gradle.develocity[latest Develocity plugin] from the Gradle Plugin Portal.
```

> More information on the compatibility can be found link:https://gradle.com/help/gradle-plugin-compatibility/[here].

Reviewer 2: ```suggestion
See the link:https://gradle.com/help/gradle-plugin-compatibility/[Develocity plugin compatibility matrix] for details.
```

Author: Thanks a lot :) @lkasso

> More information on the compatibility can be found link:https://docs.gradle.com/develocity/compatibility/#build_scans[here].

Reviewer: The link is not good. We have some stable redirects in cloudflare, that always point to the right location.
```suggestion
More information on the compatibility can be found link:https://gradle.com/help/gradle-plugin-compatibility/[here].
```
